src/blockdata/locktime.rs

```diff
@@ -0,0 +1,760 @@
+///
+/// [Bitcoin Core]: https://github.com/bitcoin/bitcoin/blob/9ccaee1d5e2e4b79b0a7c29aadb41b97e4741332/src/script/script.h#L39
+pub const LOCK_TIME_THRESHOLD: u32 = 500_000_000;
+
+/// Packed lock time wraps a [`LockTime`] consensus value i.e., the raw `u32` used by the network.
+///
+/// This type implements an naive ordering based on the `u32`, this is _not_ a semantically correct
+/// ordering for a lock time, hence [`LockTime`] does not implement `Ord`. This type is useful if
+/// you want to use a lock time as part of a struct and wish to derive `Ord`. For all other uses,
+/// consider using [`LockTime`] directly.
+///
+/// # Examples
+/// ```
+/// # use bitcoin::{Amount, PackedLockTime, LockTime};
+/// #[derive(Copy, Clone, PartialEq, Eq, PartialOrd, Ord, Debug, Hash)]
+/// struct S {
+///     lock_time: PackedLockTime,
+///     amount: Amount,
+/// }
+///
+/// let _ = S {
+///     lock_time: LockTime::from_consensus(741521).into(),
+///     amount: Amount::from_sat(10_000_000),
+/// };
+/// ```
+#[derive(Copy, Clone, PartialEq, Eq, PartialOrd, Ord, Debug, Hash)]
+#[cfg_attr(feature = "serde", derive(Serialize, Deserialize))]
+#[cfg_attr(feature = "serde", serde(crate = "actual_serde"))]
+pub struct PackedLockTime(pub u32);
+
+impl PackedLockTime {
+    /// If [`crate::Transaction::lock_time`] is set to zero it is ignored, in other words a
+    /// transaction with nLocktime==0 is able to be included immediately in any block.
+    pub const ZERO: PackedLockTime = PackedLockTime(0);
+
+    /// Returns the inner `u32`.
+    #[inline]
+    pub fn to_u32(&self) -> u32 {
+        self.0
+    }
+}
+
+impl fmt::Display for PackedLockTime {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        fmt::Display::fmt(&self.0, f)
+    }
+}
+
+impl Encodable for PackedLockTime {
+    #[inline]
+    fn consensus_encode<W: Write + ?Sized>(&self, w: &mut W) -> Result<usize, io::Error> {
+        self.0.consensus_encode(w)
+    }
+}
+
+impl Decodable for PackedLockTime {
+    #[inline]
+    fn consensus_decode<R: Read + ?Sized>(r: &mut R) -> Result<Self, encode::Error> {
+        u32::consensus_decode(r).map(PackedLockTime)
+    }
+}
+
+impl From<LockTime> for PackedLockTime {
+    fn from(n: LockTime) -> Self {
+        PackedLockTime(n.to_consensus_u32())
+    }
+}
+
+impl From<PackedLockTime> for LockTime {
+    fn from(n: PackedLockTime) -> Self {
+        LockTime::from_consensus(n.0)
+    }
+}
+
+impl From<&LockTime> for PackedLockTime {
+    fn from(n: &LockTime) -> Self {
+        PackedLockTime(n.to_consensus_u32())
+    }
+}
+
+impl From<&PackedLockTime> for LockTime {
+    fn from(n: &PackedLockTime) -> Self {
+        LockTime::from_consensus(n.0)
+    }
+}
+
+impl From<PackedLockTime> for u32 {
+    fn from(p: PackedLockTime) -> Self {
+        p.0
+    }
+}
+
+impl FromStr for PackedLockTime {
+    type Err = ParseIntError;
+
+    fn from_str(s: &str) -> Result<Self, Self::Err> {
+        s.parse().map(PackedLockTime)
```

Comment:
Just realized this doesn't support `0xffff` form which we may want to support eventually. Not blocking.

Reply:
Added an issue: https://github.com/rust-bitcoin/rust-bitcoin/issues/1112